### PR TITLE
Update usage of Stylelint, although without moving to v16

### PR DESCRIPTION
**Pull request by zglicz**
Improves #4481

After an initial investigation, turns out Styleint requires esmodules and is incompatible with CommonJs. Let us for now improve the usage of the plugins, as this would also be enforced in the new version of Stylelint - https://github.com/stylelint/stylelint/blob/46f5eaa7c464b56df1a37e516e21d0c64d5058e5/types/stylelint/index.d.ts#L252

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -22,6 +22,9 @@ sonarjs-*
 # Visual Studio
 .vs/
 
+# VS Code
```
Comment (zglicz) [inline .gitignore:25]: Removing VSCode, as if one wants to run Sonar with Connected-Mode the .vscode/settings.json would be different.

File: .vscode/settings.json [hunk 2]
```diff
@@ -1,9 +0,0 @@
-{
-  "[typescript]": {
-    "editor.defaultFormatter": "esbenp.prettier-vscode",
-    "editor.formatOnSave": true
-  },
-  "sonarlint.connectedMode.project": {
```
Comment (zglicz) [inline .vscode/settings.json:6]: there is a key "connectionId" that links to a sonarqube instance

File: packages/css/src/linter/wrapper.ts [hunk 3]
```diff
@@ -56,18 +48,6 @@ export class LinterWrapper {
...
-
-  /**
-   * Defines the wrapper's rules
-   *
-   * Besides Stylelint rules, the linter wrapper includes all the rules that
-   * are implemented internally.
-   */
-  private defineRules() {
-    for (const key in rules) {
-      stylelint.rules[key] = rules[key];
```
Comment (zglicz) [inline packages/css/src/linter/wrapper.ts:68]: here we change the internals of styleline, which is impossible in the next version.

File: packages/css/src/rules/S125/rule.ts [hunk 4]
```diff
@@ -22,7 +22,7 @@
 import * as stylelint from 'stylelint';
 import { parse } from 'postcss';
 
-const ruleName = 'no-commented-code';
+const ruleName = 'sonar/no-commented-code';
```
Comment (zglicz) [inline packages/css/src/rules/S125/rule.ts:25]: Just leaving a comment, that calling styelint.createPlugin doesn't actually register a plugin it only returns an object in a proper format. To correctly introduce a plugin, we need to add ```plugins``` as an option to the linter (as we now do in wrapper.ts)

**Review by vdiez — APPROVED** [review]
LGTM!

**Pull request merged**